lib/scoped_search/query_builder.rb

```diff
@@ -450,7 +450,9 @@ def to_sql(builder, definition, &block)
           field = definition.field_by_name(value)
           if field && field.set? && field.complete_value.values.include?(true)
             key = field.complete_value.map{|k,v| k if v == true}.compact.first
-            return builder.set_test(field, :eq, key, &block)
+            sql, *params = builder.set_test(field, :eq, key, &block)
+            params.each { |p| yield(:parameter, p) }
```

Comment:
Looking at the comment for this method:
```ruby
# for boolean fields allow a short format (example: for 'enabled = true' also allow 'enabled')
```
I'd assume this should work already without the patch. Is this patch needed because it doesn't work for renamed fields or if the values are the same as the name of the complete values?

Also, how this particular line changes something? I mean, `sql` is not changed here, the result from yielding is not stored. Is there some dark-side-effect magic?

P.S. Please bear with me, I'm not so familiar with the library.

Reply:
When building a sql query, the query builder walks the ast of the given search query by calling `#to_sql` and while the sql is assembled as the tree is being walked and then returned as a single string containing placeholders, values for placeholders are yielded into the block that is passed to `#to_sql`. The block then stores them in local variables inside the method from which the root `#to_sql` is called.

> the result from yielding is not stored. Is there some dark-side-effect magic?

Very much, here's where most of it is happening[1]. The result of yielding is not stored, but yielding a value makes the value get stored.

> Is this patch needed because it doesn't work for renamed fields or if the values are the same as the name of the complete values?

Iirc it doesn't work for any. It correctly detects that it we're trying to search by a field no matter if we're using the normal or renamed name. It generates a sql query containing a placeholder, but it does not yield any value to fill it.

[1] - https://github.com/wvanbergen/scoped_search/blob/master/lib/scoped_search/query_builder.rb#L64